lib/spack/spack/spec.py

```diff
@@ -3946,6 +3946,14 @@ def __init__(self):
 _lexer = SpecLexer()
 
 
+def add_platform(spec):
```

Comment:
Other methods of this sort (add_depedency, etc) are methods of the Spec instance. I think we should probably move this to mirror those.

Also, I would rename this to `Spec.add_default_platform` for clarity.

Reply:
addressed.